app/controllers/api/predictions_controller.rb

```diff
@@ -0,0 +1,54 @@
+module Api
+  class PredictionsController < ApplicationController
+    PREDICTIONS_LIMIT = 1000
+
+    before_filter :authenticate_by_api_token
+    before_filter :build_predictions, only: [:index]
+
+    def index
+      render json: @predictions, status: 200
+    end
+
+    def create
+      @prediction = build_new_prediction
+
+      if @prediction.save
+        render json: @prediction, status: 200
+      else
+        render json: @prediction.errors, status: 422
+      end
+    end
+
+    private
+
+    def authenticate_by_api_token
+      @user = User.find_by_api_token(params[:api_token]) rescue nil
+
+      if @user.nil? || params[:api_token].nil?
+        render json: invalid_message, status: 401
```

Comment:
Convention is to use the named statuses. See http://guides.rubyonrails.org/layouts_and_rendering.html#the-status-option